control-plane/api/mesh/v2beta1/gateway_class_types.go

```diff
@@ -37,7 +24,7 @@ type GatewayClass struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   pbmesh.GatewayClass `json:"spec,omitempty"`
+	Spec   GatewayClassSpec `json:"spec,omitempty"`
```

Comment:
This is the crux of the whole PR where we no longer consume the proto type from Consul